### PR TITLE
feat(statistical-detectors): Store states in redis

### PR DESCRIPTION
**Pull request by Zylphrex**
The moving averages need to be stored so they can be updated in the next iteration. Here we choose to start with Redis to store the moving averages.

### CODE REVIEW

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -118,6 +118,7 @@ def env(
 SENTRY_INTEGRATION_ERROR_LOG_REDIS_CLUSTER = "default"
 SENTRY_DEBUG_FILES_REDIS_CLUSTER = "default"
 SENTRY_MONITORS_REDIS_CLUSTER = "default"
+SENTRY_STATISTICAL_DETECTORS_REDIS_CLUSTER = "default"
```
Comment (Zylphrex) [inline src/sentry/conf/server.py:121]: TODO: this needs to be broken out to a separate PR and updated in prod settings before it is used

File: src/sentry/statistical_detectors/detector.py [hunk 2]
```diff
@@ -70,80 +75,8 @@ def from_dict(d: Any) -> TrendState:
 
 @dataclass
 class TrendPayload:
+    project_id: int
```
Comment (volokluev) [inline src/sentry/statistical_detectors/detector.py:79]: if the `project_id` is in the key necessary to retrieve the trend payload, why does it need to be in the value as well?

**Review by volokluev — COMMENTED** [review]
I'm having trouble reading through this PR and understanding how the pieces fit together. I think this is due to a lack of a unifying interface for this functionality. Here's one way I could imagine it looking

```python3
analyzer = EMATrendAnalyzer(
    min_data_points, 
    # other configuration things
)
analyzer.update(timeseries_name, data_point)
analyzer.get_trend(timeseries_name)
```

I'm showing the example not to say it's the "right" way of doing things, in fact I'm sure you can come up with something better. I do however want to achieve the following:

* storage layer agnosticism: if we want to switch to using the metrics platform for this, we can without rewriting the internals
* clarity of interface: as a reader of this code, I want there to be a digestsible top-level module with understandable behavior that I can drill down to if I need to. Presently It's hard to know where this code begins and ends
* clarity in configuration vs logic: There are tunable parameters in this system (e.g. `min_data_points`) which make up the configuration of this component. Being able to define them up front makes it easier to reason about how this system's functionality can be changed.

File: src/sentry/statistical_detectors/processing.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    ema_short.update(payload.value)
+
+    # This EMA uses a longer period to follow the overal trend of the timeseries.
+    ema_long = ExponentialMovingAverage(smoothing=2, period=40)
+    ema_long.set(state.long_ma, state.count)
+    ema_long.update(payload.value)
+
+    # The heuristic isn't stable initially, so ensure we have a minimum
+    # number of data points before looking for a regression.
+    stablized = state.count > MIN_DATA_POINTS
```
Comment (udameli) [inline src/sentry/statistical_detectors/processing.py:70]: 👍

**Review by udameli — APPROVED** [review]
lgtm

**Review by volokluev — APPROVED** [review]
Nice! much easier to follow. Great job Tony

**Review by udameli — APPROVED** [review]
